src/NodeWrapper.ts

```diff
@@ -472,4 +474,86 @@ export default class NodeWrapper extends SelectableObject {
     this.nodeAcceptCircle.stroke(StateManager.colorScheme.nodeAcceptStrokeColor);
     this.nodeLabel.fill(StateManager.colorScheme.nodeLabelColor);
   }
+
+  public setErrorState(isError: boolean) {
+    this._isErrorState = isError;
+    this.updateAppearance();
+}
+
+
+private updateAppearance() {
+  // Remove existing error elements (if they exist) to avoid duplicates
+  this.nodeGroup.find('.errorIcon, .errorText').forEach((el) => el.destroy());
+
+  if (this._isErrorState) {
+      // Error state appearance
+      this.nodeBackground.fill('rgb(254, 226, 225)'); // Light red
```

Comment:
Rather than being hardcoded into the `NodeWrapper` methods, these colors should likely be a part of the color schemes, so that they can adapt to light and dark mode.